float64 msg_pos_x
float64 msg_pos_y
float64 msg_vel_x
float64 msg_vel_y
